compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -109,20 +109,20 @@ namespace TR { class RegisterDependencyConditions; }
 
 TR_X86ProcessorInfo OMR::X86::CodeGenerator::_targetProcessorInfo;
 
-void TR_X86ProcessorInfo::initialize(TR::CodeGenerator *cg)
+void TR_X86ProcessorInfo::initialize()
    {
    if (_featureFlags.testAny(TR_X86ProcessorInfoInitialized))
       return;
    // For now, we only convert the feature bits into a flags32_t, for easier querying.
    // To retrieve other information, the VM functions can be called directly.
    //
-   _featureFlags.set(cg->comp()->target().cpu.getX86ProcessorFeatureFlags());
-   _featureFlags2.set(cg->comp()->target().cpu.getX86ProcessorFeatureFlags2());
-   _featureFlags8.set(cg->comp()->target().cpu.getX86ProcessorFeatureFlags8());
+   _featureFlags.set(TR::Compiler->target.cpu.getX86ProcessorFeatureFlags());
+   _featureFlags2.set(TR::Compiler->target.cpu.getX86ProcessorFeatureFlags2());
+   _featureFlags8.set(TR::Compiler->target.cpu.getX86ProcessorFeatureFlags8());
```

Comment:
@dsouzai : this reverts back to the way these feature flags used to be set before you changed it to a per-compilation thread setting in #4568.  Can you confirm this won't be problematic for what you were trying to accomplish?

Reply:
won't this cause a problem when we want to have one set of features for portable AOT and one for the code being generated to run in the current process? I thought the point of moving the stuff into the cg was to allow that variability?

Reply:
@harryyu1994 can correct me if I'm wrong, but I believe the feature flags here are only used in the "old" mechanism, which doesn't have a notion of a different target environment. I guess in that context, initializing this struct every time is unnecessary. When I went through and replaced uses of `TR:Compiler->target` with `comp->target()`, I just did so as a blanket operation. @harryyu1994 will this code eventually be deleted when the "new" mechanism becomes the default?

Reply:
Yes Irwin is right, this is the "old" mechanism and in OpenJ9 we already switched to using the "new" mechanism (TR::CPU).
The per-compilation cpu (as opposed to a global cpu) is achieved by storing "TR::CPU" objects in each compilation's target environment (`comp->target().cpu`).

Here's one example in OpenJ9, we change the target environment before each compilation. It applies to both JITServer (where Server needs to use Client's TR::CPU) and PortableAOT (where we need a portable TR::CPU):
```cpp
         // In JITServer, we would like to use JITClient's processor info for the compilation
         // The following code effectively replaces the cpu with client's cpu through the getProcessorDescription() that has JITServer support
         TR::Environment target = TR::Compiler->target;
#if defined(J9VM_OPT_JITSERVER)
         if (that->_methodBeingCompiled->isOutOfProcessCompReq())
            {
            // Customize target.cpu based on the processor description fetched from the client
            OMRProcessorDesc JITClientProcessorDesc = TR::Compiler->target.cpu.getProcessorDescription();
            target.cpu = TR::CPU::customize(JITClientProcessorDesc);
            }
         else
#endif /* defined(J9VM_OPT_JITSERVER) */
            {
            if (vm->needRelocatableTarget())
               {
               target = TR::Compiler->relocatableTarget;
               }
            }
         compiler = new (p->trMemory(), heapAlloc) TR::Compilation(
               that->getCompThreadId(),
               vmThread,
               vm,
               compilee,
               p->_ilGenRequest,
               *options,
               p->_dispatchRegion,
               p->trMemory(),
               p->_optimizationPlan,
               reloRuntime,
               &target);
```

>  I thought the point of moving the stuff into the cg was to allow that variability? @andrewcraik 

The old mechanism on x86 which exists in cg was not adopted into the design because we wanted an unified implementation across the different platforms. Power and Z already partially uses TR::CPU and that's why x86 moved to using TR::CPU as well. Also we were able to move the actual processor detection logic into omr port library with the TR::CPU design.

The old mechanism on x86 is not removed yet because the TR:CPU design needs access to omr port library and currently we do not have access to it in the compiler component. So standalone OMR still switches to the old mechanism for answers under the hood. OpenJ9 uses the new mechanism because its compiler component does have access to the omr port library.

example new mechanism API `supportsFeature()`. If no omrPortLib, calls old mechanism. 
```cpp
bool
OMR::X86::CPU::supportsFeature(uint32_t feature)
   {
   if (TR::Compiler->omrPortLib == NULL)
      return self()->supports_feature_old_api(feature);
   TR_ASSERT_FATAL(self()->supports_feature_test(feature), "old api and new api did not match, feature %d", feature);

   OMRPORT_ACCESS_FROM_OMRPORT(TR::Compiler->omrPortLib);
   return TRUE == omrsysinfo_processor_has_feature(&_processorDescription, feature);
   }
```

I hope this makes it clear.

Reply:
> will this code eventually be deleted when the "new" mechanism becomes the default?

Yes, this code will eventually be deleted when we have access to omrportlib in the OMR compiler component.